src/parsec.rs

```
@@ -366,7 +366,29 @@ impl<T: NetworkEvent, S: SecretId> Parsec<T, S> {
     /// Once the owning peer has been removed from the section (i.e. a block with payload
     /// `Observation::Remove(our_id)` has been made stable), then no further blocks will be
     /// enqueued. So, once `poll()` returns such a block, it will continue to return `None` forever.
-    pub fn poll(&mut self) -> Option<BlockGroup<T, S::PublicId>> {
+    pub fn poll(&mut self) -> Option<Block<T, S::PublicId>> {
+        let mut block_group = self.batch_poll()?;
+
+        if cfg!(any(test, feature = "testing")) && block_group.is_empty() {
+            log_or_panic!("self.consensused_blocks.front() is empty");
```

This can be replaced by simple `panic`, although I wonder why we need this at all?

---

Fair point about the `panic`, but don't you feel it's worth not just silently swallowing what would be a logic error otherwise?

---

It's pretty harmless error - we would end up returning `None` in case the block group was empty. I don't mind keeping the check if you think it's worth it.

---

Yeah, true, albeit we're breaking the documented behaviour.  I think you've convinced me though - I'll just remove it.